pydoctor/sphinx_ext/build_apidocs.py

```diff
@@ -93,9 +93,10 @@ def on_config_inited(app: Sphinx, config: Config) -> None:
         output_path = pathlib.Path(options.htmloutput)
         temp_path = output_path.with_suffix('.pydoctor_temp')
 
-        # Update intersphinx_mapping.
         pydoctor_url_path = config.pydoctor_url_path
-        if pydoctor_url_path:
+        if pydoctor_url_path and key == "main":
+            # Update intersphinx_mapping only for the main API doc.
+            # FIXME:https://github.com/twisted/pydoctor/issues/339
```

Comment:
While it's good to have a workaround, is there any reason to not fix this properly now?

If I understand it correctly, the root of the problem is that we have two projects (`main` and `epydoc_demo`) for which API docs are generated, but only one `pydoctor_url_path`. While only updating the mapping for `main` fixes the broken links in our documentation, it means we aren't able to link to anything inside the epydoc demo.

A possible solution would be to treat `pydoctor_url_path` the same as `pydoctor_args` and support both a single value and a dictionary. That shouldn't require a lot of code to implement (see line 85 above).

Reply:
I tried to reduce the scope of this PR  and at the same time, not to block this PR while #339 is not yet fixed.
So I prefer to merge it as it is :)

Reply:
Issue fixed :) Thanks for your push :p